### ISSUE TITLE
Unclear depth\stancil usage in attachment + non-attachment

### ISSUE THREAD

**Issue by krOoze** [KhronosGroup/Vulkan-Docs#1340]
This paragraph feels deeply unlclear to me:

> For depth/stencil attachments, each aspect **can** be used separately as attachments and non-attachments as long as the non-attachment accesses are also via an image subresource in either the `VK_IMAGE_LAYOUT_DEPTH_READ_ONLY_STENCIL_ATTACHMENT_OPTIMAL` layout or the `VK_IMAGE_LAYOUT_DEPTH_ATTACHMENT_STENCIL_READ_ONLY_OPTIMAL` layout, and the attachment resource uses whichever of those two layouts the image accesses do not. Use of non-attachment aspects in this case is only well defined if the attachment is used in the subpass where the non-attachment access is being made, or the layout of the image subresource is constant throughout the entire render pass instance, including the `initialLayout` and `finalLayout`.

It is hard to parse and to interpret any meaning out of this. Some specific issues:

1) "each aspect **can** be used separately as attachments and non-attachments" seems misleading as I would assume e.g. the `VK_IMAGE_LAYOUT_DEPTH_READ_ONLY_STENCIL_ATTACHMENT_OPTIMAL` cannot use the stencil as non-attachment due to the layout.

2) "attachment resource uses whichever of those two layouts" probably means aspects, not layouts (as per #1284).

3) "the image accesses do not" probably means non-attachment accesses.

4) `VK_IMAGE_LAYOUT_DEPTH_READ_ONLY_STENCIL_ATTACHMENT_OPTIMAL` is deeply confusing name. It is unclear if it means "depth read only" or "read only stencil". Also this paragraph does not even mention `VK_IMAGE_LAYOUT_DEPTH_READ_ONLY_OPTIMAL` etc. The matching rules are weird, and probably make some VUs ambiguous. It is not even clear if `oldLayout = VK_IMAGE_LAYOUT_DEPTH_READ_ONLY_STENCIL_ATTACHMENT_OPTIMAL, newLayout = VK_IMAGE_LAYOUT_DEPTH_READ_ONLY_OPTIMAL` is valid and if it constitutes a layout change or not. It is kinda its own issue, but this interface does not feel clean enough for core.

5)
> Use of non-attachment aspects in this case is only well defined if the attachment is used in the subpass where the non-attachment access is being made

Kinda negates the previous sentence that both should not access the aspect.
Not clear here what "used" means. Referenced by `pDepthStencilAttachment`?

**Comment by r-potter** [comment]
Thanks for reporting this. We have an issue for this internally and are currently discussing solutions.

**Comment by krOoze** [comment]
[Forum](https://community.khronos.org/t/spec-issue-with-vk-image-layout-depth-read-only-stencil-attachment-optimal/107403) would like to know if there are any news?

**Comment by oddhack** [comment]
This should be fixed in the 1.3.205 spec update.

**Comment by prideout** [comment]
For those who are interested, the new spec language is here:

https://github.com/KhronosGroup/Vulkan-Docs/blob/9e4c847a9e93cccd5f46e173c3ccf752b89c35df/chapters/renderpass.txt#L4099-L4121

**Comment by krOoze** [comment]
That's kinda weird. The note somewhat says the opposite of the paragraph above.
